apiserver/paasng/tests/paas_wl/resources/base/test_client.py

```diff
@@ -57,34 +57,27 @@
 
 
 class TestClientProcess:
-    @pytest.fixture
-    def app(self, set_structure, bk_stag_wl_app):
-        bk_stag_wl_app.region = RG
-        bk_stag_wl_app.name = "bk-fake-stag"
-        bk_stag_wl_app.save()
-        set_structure(bk_stag_wl_app, {"web": 2, "worker": 1})
-        return bk_stag_wl_app
-
-    @pytest.fixture
-    def release(self, app):
-        release = release_setup(
-            fake_app=app,
+    @pytest.fixture(autouse=True)
+    def setup(self, set_structure, bk_stag_wl_app):
+        self.wl_app = bk_stag_wl_app
```

Comment:
强烈不建议给 self 赋值，这代码能运行仅仅因为目前 pytest 实现如此，pytest 不提供 `self` 的状态管理
> Notice that the methods are only referencing self in the signature as a formality. No state is tied to the actual test class as it might be in the unittest.TestCase framework. Everything is managed by the pytest fixture system.

举个反例
```
class TestSelf:
    @pytest.fixture(scope="class", autouse=True)
    def fixture_1(self):
        self.a = 1
        print("1", self)

    @pytest.fixture(autouse=True)
    def fixture_2(self):
        self.b = 2
        print("2", self)

    def test_case(self):
        assert self.b == 2
        assert self.a == 1


1 <TestSelf object at 0x116894ee0>  # id 不一样的
2 <TestSelf object at 0x116859610>
self = <TestSelf object at 0x116859610>

    def test_case(self):
        assert self.b == 2
>       assert self.a == 1
E       AttributeError: 'TestSelf' object has no attribute 'a'
```

Reply:
如果是觉得原来的 bk_stag_wl_app 名字太长，可以做个 alias

```
@pytest.fixture
def wl_app(bk_stag_wl_app):
  return bk_stag_wl_app
```

项目里存量的仍然保持 self 赋值的单测是从 unittest 迁移过来没修改的, 迁移时遇到的话可以改了, 不改也行。但是增量的单测不要用这种风格。

Reply:
好的，我把这波迁移的都检查下，去掉 self.xxx 的初始化，改成 fixture